pkg/client/client.go

```diff
@@ -134,3 +134,39 @@ func (p ApplyClient) createObj(newResource runtime.Object, metaNew v1.Object, ow
 	}
 	return p.cl.Create(context.TODO(), newResource)
 }
+
+// Apply applies the objects, ie, creates or updates them on the cluster
+// returns `true, nil` if at least one of the objects was created or modified,
+// `false, nil` if nothing changed, and `false, err` if an error occurred
+func (p ApplyClient) Apply(objs []runtime.RawExtension, newLabels map[string]string) (bool, error) {
+	createdOrUpdated := false
+	for _, rawObj := range objs {
+		obj := rawObj.Object
+		if obj == nil {
+			continue
+		}
+
+		acc, err := meta.Accessor(obj)
```

Comment:
we could probably skip that part if `newLabels` is empty

Reply:
ah, I merged the PR before you added the comment - sorry for not waiting a bit longer :pray: 
I can either open another PR with the change or add the change next time when I'll be changing the common repo. 
And thanks a lot for your review :-) :+1:

Reply:
no worries, this is just an optimisation, so it can be addressed in a separate PR, indeed ;)